app/src/test/java/org/simple/clinic/enterotp/EnterOtpUiRendererTest.kt

```diff
@@ -64,4 +64,21 @@ class EnterOtpUiRendererTest {
     verify(ui).showOtpEntryMode(BruteForceOtpEntryLocked(blockedUntil))
     verify(ui).showLimitReachedError(attemptsMade)
   }
+
+  @Test
+  fun `when the otp failed attempt is more than 3, then allow otp entries with failed attempts error`() {
```

Comment:
The problem with this test is that it is dependent on the implementation detail that the maximum number of PIN attempts is 3 and that it is hardcoded in the model. This makes the test covariant with the production code and should be avoided.

The right way to fix this is to accept the entry limit as a dependency in the model and then provide it when initialising the model. In production code, we can make this part of the DI graph with a configuration object, while in tests they can be directly instantiated.

Reply:
So the maximum OTP entries is configured in the DI graph in `BruteForceOtpEntryProtectionConfigModule`, this is minimum entries for rendering error messages. So allowing and blocking is handled via the config. Do we need to add the minimumOTPEntries number in the DI graph as well? 
Refer `EnterOtpUiRenderer#generateUiForAllowingOtpEntry` for context.

Reply:
So the maximum OTP entries is configured in the DI graph in BruteForceOtpEntryProtectionConfigModule, this is minimum entries for rendering error messages. So allowing and blocking is handled via the config. Do we need to add the minimumOTPEntries number in the DI graph as well?
Refer EnterOtpUiRenderer#generateUiForAllowingOtpEntry for context.